src/states/common/approved.rs

```diff
@@ -362,13 +362,15 @@ pub trait Approved: Base {
             return Ok(());
         }
 
-        if self.peer_map().has(&their_pub_id) {
-            trace!(
-                "{} - Not sending connection request to {} - already connected.",
-                self,
-                their_pub_id
-            );
-            return Ok(());
+        if let Some(p2p_node) = self.chain().get_p2p_node(their_pub_id.name()) {
```

Comment:
Why do we need to send the connection request if we have the `P2pNode`?

Reply:
If we have their `P2pNode` we are connected, but are we also connected at the network layer?

Reply:
Not necessarily, but the whole idea behind this work is that it doesn't matter. We make no difference between knowing someones connection info and actually being connected. I think.

Reply:
(That is once it is finished, but that specific piece of work does not change the model of all node we know are actually connected with quic_p2p).